### ISSUE TITLE
Seeing unskippable ads in the beginning of videos sometimes.

### ISSUE THREAD

**Issue by brentleywilson** [revanced/revanced-patches#40]
I am not able to correctly say how often this is happening, but now all in all i have seen it occur 4 or 5 times in total till now. The ads often seems to come in pairs some going as longs as 30 seconds to a minute. Even if you were to close the video mid an ad and reopen it instantly the second ad comes to play. I was able to get an ad on opening the same video that i got an ad before by playing it after an hour or so, but then there was just a single ad not two. Almost always the ads are regional and based on language. The first ad i got was from the region of the location of my vpn at that time. The ads are always at the beginning, no ads in between a video. The app currently showing this was built using the latest cli and patch (1.4.2 and 1.6.4) and youtube version 17.22.36

**Comment by oSumAtrIX** [comment]
Unfortunately, I do not experience this. I need reproducible steps to be able to fix this issue.

**Comment by brentleywilson** [comment]
I tried going incognito on the browser and find videos that actually have ads (in the beginning) and then play them on the app, so far none of them is showing up.

**Comment by brentleywilson** [comment]
One interesting find: I got some of these ads on channels that dont meet the monetization criteria (like below 1000 subs) so is it possible that there is a different ad delivery mechanism for ads that YouTube themselves put in ?

**Comment by TheJeterLP** [comment]
> One interesting find: I got some of these ads on channels that dont meet the monetization criteria (like below 1000 subs) so is it possible that there is a different ad delivery mechanism for ads that YouTube themselves put in ?

Could be true. Weird that it only happens in some cases tho

**Comment by brentleywilson** [comment]
> Could be true. Weird that it only happens in some cases tho

Its so random, there is at least nothing on the surface that shows what is triggering it

**Comment by OxrxL** [comment]
>i found ads on the homepage : https://imgur.com/a/tFPSHdN

This could be eventually fixed easily, by adding a template check: "home_video_with_context" on this code portion:

```
if ( value.contains("related_video_with_context") && bufferBlockList .stream() .anyMatch(StandardCharsets.UTF_8.decode(buffer).toString()::contains) ) return true;
```

from file LithoAdRemoval.

in order to make the new code as follows:

```
if ( 
value.contains("home_video_with_context") && value.contains("related_video_with_context") && bufferBlockList .stream() .anyMatch(StandardCharsets.UTF_8.decode(buffer).toString()::contains) ) return true;
```

**Comment by OxrxL** [comment]
Obviously the above change need to be tested, but the bytebuffer result should be equal to the elements of bytebufferblocklist. 👍

**Comment by OxrxL** [comment]
> i found ads on the homepage : https://imgur.com/a/tFPSHdN

https://github.com/revanced/revanced-integrations/pull/24

**Comment by mataputih** [comment]
I also have the exact same problem. unskippable ads in the beginning of the video (on video click). happens usually when opening the app after a while or after watching a video for some time and then clicking on another video.

https://user-images.githubusercontent.com/65915636/176369975-c7fbc28d-d3eb-49c9-8d0d-b4f93f978c1f.mp4

**Comment by brentleywilson** [comment]
> I also have the exact same problem. unskippable ads in the beginning of the video (on video click). happens usually when opening the app after a while or after watching a video for some time and then clicking on another video.
> screen-Veed.mp4

Exactly! Its surprising that many people arent seeing this, maybe we use youtube way too much!!

**Comment by oSumAtrIX** [comment]
> Hmmm what do we have/do differently than most people? We use the same patcher, same apk version, and the same vanced microg. Maybe android version or custom rom has anything to do with it?

Its random. The patch is faulty/needs to be extended to support this ad. Unfortunately, unless we get this ad we can not fix it. So someone has to volunteer.

**Comment by OxrxL** [comment]
Without a modder who can visualize this AD, it's very difficult to be able to identify the method that displays it.

And the difficulty increases, because this isn't a litho AD.

**Comment by xemulat** [comment]
Hmmm i have the same issue, newest version built with my Packer.

**Comment by OxrxL** [comment]
> Hmmm i have the same issue, newest version built with my Packer.

Maybe it can be displayed on an emulator, with a specific region and time zone.

**Comment by xemulat** [comment]
I live on Poland, i see ads before a video

**Comment by mataputih** [comment]
I dont know if this helps (well i dont know how any of the patcher works anyway) but these went through exactly when the ad plays
![IMG_20220630_131201](https://user-images.githubusercontent.com/65915636/176663565-67e0a16d-ae55-4b1c-a292-c726eabf0e52.jpg)

**Comment by brentleywilson** [comment]
> So someone has to volunteer.

how do i do that ? i can help if its possible

**Comment by oSumAtrIX** [comment]
You will need a computer, Android Studio, an emulator, and knowdledge of debugging. If you do, ping us on Discord.

**Comment by brentleywilson** [comment]
> knowledge of debugging

:(

**Comment by mataputih** [comment]
> You will need a computer, Android Studio, an emulator, and knowdledge of debugging. If you do, ping us on Discord.

Will do when i have time

**Comment by mataputih** [comment]
Changing the country to any other country outside europe (settings -> general -> location) remove ads entirely. So probably to reproduce the problem just change the country to europe countries e.g. Germany and you will get unskippable ads

**Comment by oSumAtrIX** [comment]
Can not reproduce

**Comment by TheJeterLP** [comment]
> Changing the country to any other country outside europe (settings -> general -> location) remove ads entirely. So probably to reproduce the problem just change the country to europe countries e.g. Germany and you will get unskippable ads

My location is germany and I use ReVanced daily and never had an unskippable ad.

**Comment by ptrtypedf** [comment]
This issue seems to be happening on specific google accounts. 

Method that worked for me to fix in order:

Force stop ReVanced and MicroG and clear both of their data. Reopen ReVanced and sign out of the google account.
Then uninstall ReVanced and MicroG after signing out.

Reinstall both ReVanced and MicroG but don't sign in yet. Check to see if you're signed in after reinstalling; sign out if you are. 
Make sure when signed out in ReVanced that your original account(s) aren't still showing as 'signed out'. The sign in box should be completely blank with only "Add Account" option in MicroG and no previous history of previously signed in accounts.

Use ReVanced without an account signed in and test videos to check for ads. If after awhile you're not getting ads while signed out then follow below:

Make a new gmail account and use this new account to sign into MicroG with instead. You still shouldn't have any ads(?)

This seemed to work for me. For some reason whenever I sign in with my original google account I start getting flooded with unskippable ads. This never happened with original Vanced fwiw.

**Comment by oSumAtrIX** [comment]
I suppose this is just a temporal solution. Ideally, we will be able to find the method which shows this ad. A memory snapshot, while the ad is playing, might reveal more info on where this ad originates from.

**Comment by RFLEpicGamer** [comment]
> You will need a computer, Android Studio, an emulator, and knowdledge of debugging. If you do, ping us on Discord.

Does that mean that I have to rebuild itself from the source?

**Comment by Legitti** [comment]
Getting ads also in Spain, trying to change the country to see if it helps

**Comment by Kipptc** [comment]
Can reproduce here as well, S22 Android 12.

It seems if the video begins to autoplay from search or home, the ads are skipped then. If the video is opened before autoplay begins, ad.

As I was typing this, I blocked adservice.google.com in my pihole, and suddenly the start ads are gone?

**Comment by Sarreos** [comment]
This error occurs when an ad plays:
`2022-08-26 15:35:38.250 29981-29981/app.revanced.android.youtube E/YouTube: ERROR onPlayAd called for new ad without reset being called. Clients in incorrect state`

**Comment by oSumAtrIX** [comment]
@Sarreos please also include the stack trace and the version of the app you are using.

**Comment by RFLEpicGamer** [comment]
@Sarreos it's a bug?

**Comment by crossxover** [comment]
> 

Using Adguard for Android and adding this site has helped me block ads.

**Comment by lahdekorpi** [comment]
I'm having these as well on occasion with a version built via the CLI, manager, and prebuilt APK.
I have access to `adb logcat` and can build a debug version. Is the best way to debug this to just run with the debug enabled and paste the log or is there a better way?

**Comment by oSumAtrIX** [comment]
Fixed in commit fbc83a22d37012394311a87d27202b7cc86e047d.
